tests/test_ha.py

```diff
@@ -1967,3 +1967,26 @@ def test_is_failover_possible(self):
             with patch.object(global_config.__class__, 'is_quorum_commit_mode', PropertyMock(return_value=True)):
                 # switchover from leader when synchronous_mode = quorum
                 self.assertFalse(self.ha.is_failover_possible())  # failure, because quorum is low
+
+    def test_failover_mode_parsing(self):
+        # Test default mode
+        f = Failover.from_node(1, '{"leader": "a", "member": "b"}')
+        self.assertEqual(f.mode, 'graceful')
+
+        # Test explicit graceful mode
+        f = Failover.from_node(1, '{"leader": "a", "member": "b", "mode": "graceful"}')
+        self.assertEqual(f.mode, 'graceful')
+
+        # Test immediate mode
+        f = Failover.from_node(1, '{"leader": "a", "member": "b", "mode": "immediate"}')
+        self.assertEqual(f.mode, 'immediate')
```

Comment:
This is not the best place for this test.
I would suggest moving it to test_kubernetes.py